From 7adc392f6bedb8926d6c93423d88c758300658ce Mon Sep 17 00:00:00 2001
From: Geert Uytterhoeven <geert@linux-m68k.org>
Date: Mon, 29 Dec 2008 16:48:41 +0100
Subject: [PATCH 28/90] m68k: Atari SCC support for ST

Enable support for Atari ST, and fix 2 obvious typos in the ST support code.

Signed-off-by: Geert Uytterhoeven <geert@linux-m68k.org>
---
 drivers/char/atari_scc.c |    5 +++--
 1 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/char/atari_scc.c b/drivers/char/atari_scc.c
index 800193c..e407aa8 100644
--- a/drivers/char/atari_scc.c
+++ b/drivers/char/atari_scc.c
@@ -51,6 +51,7 @@
 
 #define CONFIG_TT_SCC     1
 #define CONFIG_FALCON_SCC 1
+#define CONFIG_ST_SCC     1
 
 #define CHANNEL_A	0
 #define CHANNEL_B	1
@@ -475,7 +476,7 @@ static int atari_st_scc_init(void)
 	request_irq(IRQ_SCCA_STAT, scc_stat_int, IRQ_TYPE_PRIO,
 		    "SCC-A status", port);
 	request_irq(IRQ_SCCA_RX, scc_rx_int, IRQ_TYPE_PRIO, "SCC-A RX", port);
-	request_irq(SCCA_SPCOND, scc_spcond_int, IRQ_TYPE_PRIO,
+	request_irq(IRQ_SCCA_SPCOND, scc_spcond_int, IRQ_TYPE_PRIO,
 		    "SCC-A special cond", port);
 	{
 		SCC_ACCESS_INIT(port);
@@ -493,7 +494,7 @@ static int atari_st_scc_init(void)
 		/* disable interrupts for this channel */
 		SCCwrite(INT_AND_DMA_REG, 0);
 		/* Set the interrupt vector */
-		SCCwrite(INT_VECTOR_REG, BVME_IRQ_SCC_BASE);
+		SCCwrite(INT_VECTOR_REG, 0x60);
 		/* Interrupt parameters: vector includes status, status low */
 		SCCwrite(MASTER_INT_CTRL, MIC_VEC_INCL_STAT);
 		SCCmod(MASTER_INT_CTRL, 0xff, MIC_MASTER_INT_ENAB);
-- 
1.6.2.3

